evm/vm/evm.py

```diff
@@ -125,10 +106,10 @@ def get_vm_class_for_block_number(cls, block_number):
         """
         Return the vm class for the given block number.
         """
-        range = find_range(tuple(cls.vms_by_range.keys()), block_number)
-        base_vm_class = cls.vms_by_range[range]
-        vm_class = base_vm_class.configure(db=cls.db)
```

Comment:
My main gripe with this approach is that it feels unnatural to have classes configured with a DB, and also the fact that sometimes we use the configure() method and sometimes we use __init__() to set class/instance attributes. Maybe we need a clear distinction between class/instance attributes and standardise on using configure() for class attributes and __init__() for instance attributes. Something like https://github.com/pipermerriam/py-evm/pull/15/commits/a1752ccd2f89a686d8a63b1e65e77a327e7c2e1d